core/src/main/java/com/dtolabs/rundeck/plugins/jobs/JobExecutionEventImpl.java

```diff
@@ -3,36 +3,49 @@
 import com.dtolabs.rundeck.core.common.INodeSet;
 import com.dtolabs.rundeck.core.execution.ExecutionContextImpl;
 import com.dtolabs.rundeck.core.execution.ExecutionLogger;
+import com.dtolabs.rundeck.core.execution.ExecutionReference;
 import com.dtolabs.rundeck.core.execution.workflow.StepExecutionContext;
 import com.dtolabs.rundeck.core.jobs.JobExecutionEvent;
+import com.dtolabs.rundeck.core.execution.workflow.WorkflowExecutionResult;
+import com.dtolabs.rundeck.core.jobs.JobEventResult;
+import lombok.Getter;
+import lombok.Setter;
 
 import java.util.Map;
 
 public class JobExecutionEventImpl implements JobExecutionEvent {
 
-    private StepExecutionContext executionContext;
+    @Getter @Setter private StepExecutionContext executionContext;
+    @Getter @Setter ExecutionReference execution;
+    @Getter @Setter JobEventResult result;
 
-    @Override
-    public StepExecutionContext getExecutionContext() {
-        return this.executionContext;
-    }
-
-    public void setExecutionContext(StepExecutionContext executionContext) {
-        this.executionContext = executionContext;
-    }
 
     public JobExecutionEventImpl(StepExecutionContext executionContext) {
         if(executionContext != null){
             this.executionContext = ExecutionContextImpl.builder(executionContext).build();
         }
     }
 
-    public JobExecutionEventImpl() {
+    public JobExecutionEventImpl(StepExecutionContext executionContext, ExecutionReference execution) {
+        this.executionContext = executionContext;
```

Comment:
I'd prefer not to use the reference directly. Send a new one instead and the merge with the original.
```suggestion
        this.executionContext = ExecutionContextImpl.builder(executionContext).build();
```

Reply:
ok, won't hurt to do it that way